src/zino/api/legacy.py

```diff
@@ -619,3 +620,21 @@ async def do_raiseerror(self):
         that exceptions that go unhandled by a command responder is handled by the protocol engine.
         """
         1 / 0  # noqa
+
+    @requires_authentication
+    @Zino1ServerProtocol._translate_case_id_to_event
+    async def do_deleteevent(self, event: Event):
+        """Implements an DELETEEVENT command that did not exist in the Zino 1 protocol. This is just used for testing
+        the frontend.
+        """
+        events = self._state.events
+
+        index = EventIndex(event.router, event.subindex, type(event))
+
+        if events._events_by_index.get(index) and event.id == events._events_by_index[index].id:
+            del events._events_by_index[index]
+        if events._closed_events_by_index.get(index) and event.id == events._closed_events_by_index[index].id:
+            del events._closed_events_by_index[index]
```

Comment:
If the point of the first check is to see if the dict has a value for a certain index then I think this is a better way of writing it:
```suggestion
        if index in events._events_by_index and event.id == events._events_by_index[index].id:
            del events._events_by_index[index]
        if index in events._closed_events_by_index and event.id == events._closed_events_by_index[index].id:
            del events._closed_events_by_index[index]
```

Reply:
Thank you, very good point!